.gitignore

```diff
@@ -1 +1,107 @@
+ld products...
```

Comment:
do we need the '.gitignore' file at top level here?  some stuff might be irrelevant for others. 
So locally we could use global gitignore as well .

Reply:
This is copied over from ansible/ansible. Having a comprehensive gitignore saves everyone the trouble of updating it locally.